aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -291,6 +292,13 @@ Tensor & _index_put_impl_(Tensor & self, TensorList indices, const Tensor & valu
       index_put_accum_stub(self.device().type(), self, indices, value, unsafe);
       return self;
   }
+
+  if (at::has_internal_overlap(self) == MemOverlap::YES) {
+    TORCH_WARN("Use of index_put_ on expanded tensors is deprecated. "
+               "Please clone() the tensor before performing this operation.");
```

Comment:
Another way to get here is via advanced indexing. We should talk about that as well in the error message so that the user can connect these two ops. "Use of index_put_ on expanded tensors is deprecated. Please clone() the tensor before performing this operation. You may also encounter this error message when assigning items to a tensor using advanced indexing, that is also deprecated"

Reply:
Good idea. Added a note in the warning for `index_put_` and also `masked_fill_` as well.